src/index.ts

```diff
@@ -1,10 +1,17 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
+import { createIoCContainer } from './ioc';
+import type { Users } from './services/users';
+import type { User } from './types';
+const ioc = createIoCContainer();
 
-import type { User, ApiConfig } from './types';
+window.onload = (event: Event) => {
+  const logger = ioc.resolve('logger');
+  logger.info('Page is loaded.');
+
+  const usersService = ioc.resolve('users');
+  renderUsers(usersService);
```

Comment:
you shouldn't use renderUsers before it was defined. Please do not change the order of functions.